auction-server/src/opportunity/entities/opportunity_evm.rs

```diff
@@ -34,3 +69,158 @@ impl Deref for OpportunityEvm {
+
+    fn try_from(
+        val: repository::Opportunity<repository::OpportunityMetadataEvm>,
+    ) -> Result<Self, Self::Error> {
+        let sell_tokens = serde_json::from_value(val.sell_tokens.clone()).map_err(|e| {
+            tracing::error!(
+                "Failed to deserialize sell_tokens for database opportunity evm: {:?} - {}",
+                val,
+                e
+            );
+            anyhow::anyhow!(e)
+        })?;
+        let buy_tokens = serde_json::from_value(val.buy_tokens.clone()).map_err(|e| {
+            tracing::error!(
+                "Failed to deserialize buy_tokens for database opportunity evm: {:?} - {}",
+                val,
+                e
+            );
+            anyhow::anyhow!(e)
+        })?;
+        Ok(OpportunityEvm {
+            core_fields:       OpportunityCoreFields {
+                id: val.id,
+                creation_time: val.creation_time.assume_utc().unix_timestamp_nanos(),
+                permission_key: PermissionKey::from(val.permission_key),
+                chain_id: val.chain_id,
+                sell_tokens,
+                buy_tokens,
+            },
+            target_contract:   val.metadata.target_contract,
+            target_call_value: val.metadata.target_call_value,
+            target_calldata:   val.metadata.target_calldata.clone(),
+        })
+    }
+}
+
+impl From<OpportunityCreateEvm> for OpportunityEvm {
+    fn from(val: OpportunityCreateEvm) -> Self {
+        let id = Uuid::new_v4();
+        let odt = OffsetDateTime::now_utc();
+        OpportunityEvm {
+            core_fields:       OpportunityCoreFields::<TokenAmountEvm> {
+                id,
+                creation_time: odt.unix_timestamp_nanos() / 1000 as UnixTimestampMicros,
+                permission_key: val.core_fields.permission_key.clone(),
+                chain_id: val.core_fields.chain_id.clone(),
+                sell_tokens: val.core_fields.sell_tokens.clone(),
+                buy_tokens: val.core_fields.buy_tokens.clone(),
+            },
+            target_contract:   val.target_contract,
+            target_call_value: val.target_call_value,
+            target_calldata:   val.target_calldata.clone(),
+        }
+    }
+}
+
+impl From<OpportunityEvm> for OpportunityCreateEvm {
+    fn from(val: OpportunityEvm) -> Self {
+        OpportunityCreateEvm {
+            core_fields:       OpportunityCoreFieldsCreate::<TokenAmountEvm> {
+                permission_key: val.core_fields.permission_key.clone(),
+                chain_id:       val.core_fields.chain_id.clone(),
+                sell_tokens:    val.core_fields.sell_tokens.clone(),
+                buy_tokens:     val.core_fields.buy_tokens.clone(),
+            },
+            target_contract:   val.target_contract,
+            target_call_value: val.target_call_value,
+            target_calldata:   val.target_calldata.clone(),
+        }
+    }
+}
+
+impl PartialEq<OpportunityCreateEvm> for OpportunityEvm {
```

Comment:
with the conversions we have, we don't need this. we can convert `OpportunityEvm` into `OpportunityCreateEvm` and then compare